p2p/transport/tcp/conn/connection.go

```diff
@@ -655,30 +659,33 @@ FOR_LOOP:
 		case *tmp2p.Packet_PacketMsg:
 			channelID := byte(pkt.PacketMsg.ChannelID)
 			channel, ok := c.channelsIdx[channelID]
-			if pkt.PacketMsg.ChannelID < 0 || pkt.PacketMsg.ChannelID > math.MaxUint8 || !ok || channel == nil {
+			if !ok || pkt.PacketMsg.ChannelID < 0 || pkt.PacketMsg.ChannelID > math.MaxUint8 {
 				err := fmt.Errorf("unknown channel %X", pkt.PacketMsg.ChannelID)
-				c.Logger.Debug("Connection failed @ recvRoutine", "conn", c, "err", err)
-				c.stopForError(err)
+				c.Logger.Error("Connection failed @ recvRoutine", "err", err)
+				c.Close(err.Error())
 				break FOR_LOOP
 			}
 
 			msgBytes, err := channel.recvPacketMsg(*pkt.PacketMsg)
 			if err != nil {
-				if c.IsRunning() {
-					c.Logger.Debug("Connection failed @ recvRoutine", "conn", c, "err", err)
-					c.stopForError(err)
-				}
+				c.Logger.Error("Connection failed @ recvRoutine", "err", err)
```

Comment:
Do you know under which scenarios was it ok not to stop the peer here ? It was stopped only when the inner if was tru ?

Reply:
Also the log level was debug before.

Reply:
good call 👍